tests/CMakeLists.txt

```
@@ -32,10 +53,39 @@ function(add_test_suite suite_name)
         set(data_name ${suite_name})
     endif()
 
+    if(data_name STREQUAL "psa_crypto_not_supported.generated" OR
+       data_name STREQUAL "psa_crypto_storage_format.current" OR
+       data_name STREQUAL "psa_crypto_storage_format.v0")
+        set(data_file
+            ${CMAKE_CURRENT_BINARY_DIR}/suites/test_suite_${data_name}.data)
+    else()
+        set(data_file
+            ${CMAKE_CURRENT_SOURCE_DIR}/suites/test_suite_${data_name}.data)
+    endif()
+
     add_custom_command(
-        OUTPUT test_suite_${data_name}.c
-        COMMAND ${MBEDTLS_PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/scripts/generate_test_code.py -f ${CMAKE_CURRENT_SOURCE_DIR}/suites/test_suite_${suite_name}.function -d ${CMAKE_CURRENT_SOURCE_DIR}/suites/test_suite_${data_name}.data -t ${CMAKE_CURRENT_SOURCE_DIR}/suites/main_test.function -p ${CMAKE_CURRENT_SOURCE_DIR}/suites/host_test.function -s ${CMAKE_CURRENT_SOURCE_DIR}/suites --helpers-file ${CMAKE_CURRENT_SOURCE_DIR}/suites/helpers.function -o .
-        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/scripts/generate_test_code.py ${mbedtls_target} ${CMAKE_CURRENT_SOURCE_DIR}/suites/helpers.function ${CMAKE_CURRENT_SOURCE_DIR}/suites/main_test.function ${CMAKE_CURRENT_SOURCE_DIR}/suites/host_test.function ${CMAKE_CURRENT_SOURCE_DIR}/suites/test_suite_${suite_name}.function ${CMAKE_CURRENT_SOURCE_DIR}/suites/test_suite_${data_name}.data
+        OUTPUT
```

The relationship between the input arguments to `generate_test_code.py` and the output filename is non-obvious (especially since `${suite_name}` and `${data_name}` are sometimes the same).
Could you add a comment here explaining that the output filename is derived from the `-d` switch (rather than the `-f` switch, for example)?
```suggestion
        OUTPUT
        # The output filename of generate_test_code.py is derived from the -d input argument.
```